src/odemis/acq/fastem.py

```diff
@@ -712,13 +712,35 @@ def pre_calibrate(self, pre_calibrations):
                                               self._se_detector, self._ebeam_focus,
                                               calibrations=pre_calibrations)
 
+        logging.debug(f"Will run pre-calibrations at field index {self.field_idx}")
         try:
-            self._pre_calibrations_future.result()  # wait for the calibrations to be finished
-        except CancelledError:
-            logging.debug("Cancelled acquisition pre-calibrations.")
-            raise
+            for i in range(3):  # try running the pre-calibrations 3 times
+                try:
+                    self._pre_calibrations_future = align(self._scanner, self._multibeam,
```

Comment:
It is set in line 708, do we redefine it?

Reply:
I second this

Reply:
yes, actually I need to remove it from line 708, because it will not re-run the future if I just call `.result()` on the old future.